tensorzero-core/src/tool.rs

```diff
@@ -40,20 +45,54 @@ use crate::{
  * If we are doing an implicit tool call for JSON schema enforcement, we can use the compiled schema from the output signature.
  */
 
+#[derive(ts_rs::TS, AsRefStr, Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[strum(serialize_all = "snake_case")]
+#[cfg_attr(feature = "pyo3", pyclass(str))]
+pub enum Tool {
```

Comment:
Can you add docstrings on what this type is and how it differs from ClientSideFunctionTool?

Reply:
Also let's merge this + my Python PRs before the next release - I want to make these classes all dataclasses instead of PyO3 objects.